onnxruntime/core/mlas/lib/platform.cpp

```diff
@@ -464,6 +472,22 @@ Return Value:
         this->ConvSymS8S8Dispatch = &MlasConvSymS8DispatchDot;
     }
 
+    //
+    // Check if the processor supports ASIMD I8MM instructions.
+    //
+
+    bool HasI8MMInstruction = false;
+
+    uint64_t isar1_el1;
+    asm("mrs %[reg], ID_AA64ISAR1_EL1\n" : [reg] "=r"(isar1_el1) : :);
```

Comment:
Reading ID_AA64ISAR1_EL1 will definitely cause a crash on an ARMv8-A chip and potentially cause a crash on later ARMv8 chips. See https://github.com/microsoft/onnxruntime/issues/17647 which we believe is due to reading that register.

Additionally I believe EL1 registers requires additional privileges to be read safely: https://github.com/golang/go/issues/57336#issuecomment-1353800393

Any reason we can't use the value from cpuinfo which is [read using sysctl](https://github.com/pytorch/cpuinfo/blob/959002f82d7962a473d8bf301845f2af720e0aa4/src/arm/mach/init.c#L381-L384)? Would need to expose it via CPUIDInfo as we do with other values like [HasArmNeoDot](https://github.com/microsoft/onnxruntime/blob/60f19ab001b40d8e0bd2912645afba28d2c016ba/onnxruntime/core/common/cpuid_info.h#L30) which is set using cpuinfo [here](https://github.com/microsoft/onnxruntime/blob/60f19ab001b40d8e0bd2912645afba28d2c016ba/onnxruntime/core/common/cpuid_info.cc#L139).

Reply:
Hi @skottmckay , thanks for the review, I agree, in fact I started cleaning this up to use the library query, will post the updated version.